testing/str_test.py

```diff
@@ -12,7 +12,20 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# pylint: disable=redefined-outer-name
+# pylint: disable=redefined-outer-name,W0703
+
+import sys
+
+
+def checkequal(expected, s, attr, *args):
```

Comment:
Seems like this method doesn't make the tests any more readable than just writing out the assert statement. Compare `checkequal('123', '123', 'zfill', 2)` and `assert '123'.zfill(2) == '123'`. The latter is shorter is much more difficult to misread. I'd suggest just writing out the asserts.